x/crosschain/keeper/msg_server_refund_aborted_tx_test.go

```diff
@@ -0,0 +1,459 @@
+			InboundTxParams: &crosschaintypes.InboundTxParams{
+				TxOrigin:      validEthAddress.String(),
+				CoinType:      common.CoinType_Zeta,
+				SenderChainId: getValidEthChainID(t),
+			}},
+			"")
+		require.NoError(t, err)
+		require.Equal(t, validEthAddress, address)
+	})
+	t.Run("should return refund address if provided coin-type erc20", func(t *testing.T) {
+		validEthAddress := sample.EthAddress()
+		address, err := keeper.GetRefundAddress(crosschaintypes.CrossChainTx{
+			InboundTxParams: &crosschaintypes.InboundTxParams{
+				Sender:        validEthAddress.String(),
+				CoinType:      common.CoinType_ERC20,
+				SenderChainId: getValidEthChainID(t),
+			}},
+			"")
+		require.NoError(t, err)
+		require.Equal(t, validEthAddress, address)
+	})
+	t.Run("should return refund address if provided coin-type gas for btc chain", func(t *testing.T) {
+		validEthAddress := sample.EthAddress()
+		address, err := keeper.GetRefundAddress(crosschaintypes.CrossChainTx{
+			InboundTxParams: &crosschaintypes.InboundTxParams{
+				CoinType:      common.CoinType_Gas,
+				SenderChainId: getValidBtcChainID(),
+			}},
+			validEthAddress.String())
+		require.NoError(t, err)
+		require.Equal(t, validEthAddress, address)
+	})
+	t.Run("fail if refund address is not provided for btc chain", func(t *testing.T) {
+		_, err := keeper.GetRefundAddress(crosschaintypes.CrossChainTx{
+			InboundTxParams: &crosschaintypes.InboundTxParams{
+				CoinType:      common.CoinType_Gas,
+				SenderChainId: getValidBtcChainID(),
+			}},
+			"")
+		require.ErrorContains(t, err, "refund address is required for bitcoin chain")
+	})
+	t.Run("address overridden if optional address is provided", func(t *testing.T) {
+		validEthAddress := sample.EthAddress()
+		address, err := keeper.GetRefundAddress(crosschaintypes.CrossChainTx{
+			InboundTxParams: &crosschaintypes.InboundTxParams{
+				Sender:        sample.EthAddress().String(),
+				CoinType:      common.CoinType_ERC20,
+				SenderChainId: getValidEthChainID(t),
+			}},
+			validEthAddress.String())
+		require.NoError(t, err)
+		require.Equal(t, validEthAddress, address)
+		address, err = keeper.GetRefundAddress(crosschaintypes.CrossChainTx{
+			InboundTxParams: &crosschaintypes.InboundTxParams{
+				Sender:        sample.EthAddress().String(),
+				CoinType:      common.CoinType_Zeta,
+				SenderChainId: getValidEthChainID(t),
+			}},
+			validEthAddress.String())
+		require.NoError(t, err)
+		require.Equal(t, validEthAddress, address)
+		address, err = keeper.GetRefundAddress(crosschaintypes.CrossChainTx{
+			InboundTxParams: &crosschaintypes.InboundTxParams{
+				Sender:        sample.EthAddress().String(),
+				CoinType:      common.CoinType_Gas,
+				SenderChainId: getValidEthChainID(t),
+			}},
+			validEthAddress.String())
+		require.NoError(t, err)
+		require.Equal(t, validEthAddress, address)
+	})
+
+}
+func TestMsgServer_RefundAbortedCCTX(t *testing.T) {
```

Comment:
I think we miss a test where the creator is not authorized? If yes, we need to add one

Reply:
d1ff261d3c1d1bf7ed47cedfcdac85fd3d24159d